### PR TITLE
UF-295 - Upgrading lucene to 5.3.1

### PR DESCRIPTION
**Pull request by mrietveld**
See also: 

- https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/169
- https://github.com/uberfire/uberfire/pull/337
- https://github.com/uberfire/uberfire-extensions/pull/258
- https://github.com/droolsjbpm/kie-wb-common/pull/337
- https://github.com/droolsjbpm/jbpm-designer/pull/220
- https://github.com/droolsjbpm/jbpm-form-modeler/pull/46
- https://github.com/droolsjbpm/jbpm-console-ng/pull/395
- https://github.com/droolsjbpm/drools-wb/pull/156
- https://github.com/droolsjbpm/kie-wb-distributions/pull/261

### CODE REVIEW

**Comment by mrietveld** [comment]
jenkins, please retest!

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java [hunk 1]
```diff
@@ -177,4 +187,33 @@ protected static File createTempDirectory() throws IOException {
         return temp;
     }
 
+    public void searchFor( Index index, Query query, int expectedNumHits, Path... paths ) throws IOException {
+        final IndexSearcher searcher = ((LuceneIndex) index).nrtSearcher();
+        final TopScoreDocCollector collector = TopScoreDocCollector.create(10);
+        searcher.search(query, collector);
+        final ScoreDoc[] hits = collector.topDocs().scoreDocs;
+        // expectedNumHits of the properties files have a title containing "lucene"
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java]: You probably don't need this comment now ```searchFor(...)``` is generic.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java [hunk 2]
```diff
@@ -177,4 +187,33 @@ protected static File createTempDirectory() throws IOException {
...
+            }
+        }
+    }
+
+    public void searchFor( Query query, int expectedNumHits ) throws IOException {
+        final IndexSearcher searcher = ( (LuceneIndexManager) getConfig().getIndexManager() ).getIndexSearcher();
+        final TopScoreDocCollector collector = TopScoreDocCollector.create( 10 );
+        searcher.search( query, collector );
+        final ScoreDoc[] hits = collector.topDocs().scoreDocs;
+        // expectedHits of the properties files have a title containing "lucene"
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java]: You probably don't need this comment now ```searchFor(...)``` is generic.

**Comment by manstis** [comment]
+1 to merge (when @porcelli has commented on @mrietveld's question in https://github.com/uberfire/uberfire-extensions/pull/258)

**Pull request merged**